File: java/src/main/java/pro/wata/sdk/model/PaymentLink.java
package pro.wata.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.time.OffsetDateTime;
import java.util.List;

/** Ответ на создание/получение платёжной ссылки. */
@JsonIgnoreProperties(ignoreUnknown = true)
public record PaymentLink(
        String id,
        Double amount,
        Currency currency,
        PaymentLinkStatus status,
        String url,
        String terminalName,
        String terminalPublicId,
        OffsetDateTime creationTime,
        PaymentLinkType type,
        String orderId,
        OffsetDateTime expirationDateTime,
        Boolean isArbitraryAmountAllowed,
        List<Double> arbitraryAmounts
) {
}

File: dotnet/src/Wata.Sdk/Acquiring/Models/PaymentLink.cs
using Wata.Sdk.Models;

namespace Wata.Sdk.Acquiring.Models;

/// <summary>Платёжная ссылка (ответ на создание/поиск/получение по идентификатору).</summary>
public sealed record PaymentLink
{
    public Guid Id { get; init; }

    public decimal Amount { get; init; }

    public Currency Currency { get; init; }

    public PaymentLinkStatus Status { get; init; }

    public Uri? Url { get; init; }

    public string? TerminalName { get; init; }

    public string? TerminalPublicId { get; init; }

    public DateTimeOffset CreationTime { get; init; }

    public PaymentLinkType Type { get; init; }

    public string? OrderId { get; init; }

    public DateTimeOffset? ExpirationDateTime { get; init; }

    public bool IsArbitraryAmountAllowed { get; init; }

    /// <summary>Подсказки произвольной суммы, если IsArbitraryAmountAllowed включён.</summary>
    public IReadOnlyList<decimal>? ArbitraryAmounts { get; init; }
}
